app/helpers/application_helper.rb

```diff
@@ -258,6 +258,39 @@ def help_email_link
     link_to admin_address.to_s, "mailto:#{admin_address}"
   end
 
+  #
+  # Handles rendering of JSON to a series of nested lists. Does the following:
+  # String: Rendered as-is
+  # Array: Unordered list (Strictly speaking arrays are ordered, but we probably don't care.)
+  # Object: Descriptive list
+  # Other: Calls to_s
+  # Processes each in turn and clled recursively
+  #
+  # @param [Hash, String, Array,, #to_s] json The Object to render
+  #
+  # @return [String] HTML formatted for rendering
+  #
+  def render_parsed_json(json)
+    case json
+    when String; then json
+    when Array
+      tag.ul do
+        json.each { |elem, _string| concat tag.li(render_parsed_json(elem)) }
```

Comment:
`tag.li(render_parsed_json(elem))` should return a list element, right, so what does the `concat` do?

Reply:
tag.li generates the HTML as a string, concat actually pushes the tag to the page within the ul.